packages/common/src/services/integrations/jwp/JWPCheckoutService.ts

```diff
@@ -99,9 +99,12 @@ export default class JWPCheckoutService extends CheckoutService {
       customerCurrency: offer.currency,
       offerTitle: offer.description,
       active: true,
-      period: offer.access_type.period === 'month' && offer.access_type.quantity === 12 ? 'year' : offer.access_type.period,
-      freePeriods: offer.trial_period ? 1 : 0,
+      period:
+        offer.access_type.period === 'month' && offer.access_type.quantity === 12 ? 'year' : (offer.access_type.period as 'day' | 'week' | 'month' | 'year'),
       planSwitchEnabled: offer.item.plan_switch_enabled ?? false,
+
+      // trial period is always assumed to be in days
+      freeDays: offer.trial_period?.quantity || 0,
```

Comment:
Is this compatible with Cleeng offers? The `freePeriods` is a property in the Cleeng Offer response and JWP currently maps to that data structure 😬

Reply:
@mirovladimitrovski could you check this one?

Reply:
I'm not sure I follow. The Cleeng offers implementation is intact.

Reply:
Than all is good! The diff is a bit weird indeed, I thought you removed the `freePeriods` property completely.